src/object/notification.c

```diff
@@ -101,16 +95,6 @@ void sendSignal(notification_t *ntfnPtr, word_t badge)
                     MCS_DO_IF_SC(tcb, ntfnPtr, {
                         possibleSwitchTo(tcb);
                     })
-#ifdef CONFIG_KERNEL_MCS
-                    if (tcb->tcbSchedContext != NULL && sc_active(tcb->tcbSchedContext)) {
-                        sched_context_t *sc = SC_PTR(notification_ptr_get_ntfnSchedContext(ntfnPtr));
-                        if (tcb->tcbSchedContext == sc && sc_sporadic(sc)) {
```

Comment:
I still need to figure out some way handle this case. The refill_unblock_check should happen in possibleSwitchTo unless the vm thread was in the run queue with an SC that is not the current SC so isn't being unblocked at this point.  However this code isn't under verification and  doRemoteVMCheckBoundNotification doesn't even donate an SC at the moment on SMP (but it should).  So I'm leaving it for now until there's consensus whether we want to make the rest of the changes in this commit now or later.